modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2491,6 +2491,44 @@ def _callable_func(self, func, axis, *args, **kwargs):
     groupby_prod = GroupByReduce.register("prod")
     groupby_sum = GroupByReduce.register("sum")
 
+    def _mean_agg_map(dfgb, **kwargs):
+        kwargs["min_count"] = 1
+        result = dfgb.sum(**kwargs)
+        divisor = dfgb.count()
+        divisor.set_axis(
+            ["__mean_agg_size_column__" + x for x in divisor.columns],
+            axis=1,
+            inplace=True,
+        )
+        result = pandas.concat([result, divisor], axis=1, copy=False)
+        return result
+
+    def _mean_agg_reduce(dfgb, **kwargs):
+        kwargs["min_count"] = 1
+        result = dfgb.sum(**kwargs)
+        divirgent = result[
+            [x for x in result.columns if not x.startswith("__mean_agg_size_column__")]
+        ]
+        divisor = result[
+            [x for x in result.columns if x.startswith("__mean_agg_size_column__")]
+        ]
+        divisor.set_axis(
+            [x[len("__mean_agg_size_column__") :] for x in divisor.columns],
+            axis=1,
+            inplace=True,
+        )
```

Comment:
The order of columns among dividend and divisor frames should be the same, so can we just do: `divisor.columns = divirgent.columns.copy()`?

Reply:
Except divisor and divirgent have different sets of columns because divisor also has "by" columns prefixed with "__mean_agg_size_column__". The following line (with comment above it) makes sure that they are excluded.